src/main/java/mat/dao/clause/impl/MeasureDAOImpl.java

```diff
@@ -555,19 +574,20 @@ private Predicate buildPredicateForMeasureSearch(String userId, CriteriaBuilder
 
     private List<MeasureShareDTO> getOrderedDTOListFromMeasureResults(MeasureSearchModel measureSearchModel, User user,
                                                                       List<Measure> measureResultList) {
-
         final ArrayList<MeasureShareDTO> orderedDTOList = new ArrayList<>();
 
         final Map<String, MeasureShareDTO> measureSetIdDraftableMap = new HashMap<>();
 
         final List<Measure> measureSets = getAllMeasuresInSet(measureResultList);
+        final Set<Measure> measureResultSet = measureResultList.stream().collect(Collectors.toSet());
 
         for (final Measure measure : measureSets) {
             final MeasureShareDTO dto = extractDTOFromMeasure(measure);
             if (dto.isDraft()) {
                 measureSetIdDraftableMap.put(dto.getMeasureSetId(), dto);
             }
-            if (measureResultList.contains(measure)) {
+            if (measureResultSet.contains(measure)) {
+                measureResultSet.add(measure);
```

Comment:
What's the intent in calling `add` when the measure is already contained in the set?

Reply:
that's a good catch. it's not needed